lib/Parser/Scan.cpp

```diff
@@ -668,7 +669,11 @@ typename Scanner<EncodingPolicy>::EncodedCharPtr Scanner<EncodingPolicy>::FScanN
     else
     {
 LFloat:
-        *pdbl = Js::NumberUtilities::StrToDbl(p, &pchT, likelyInt);
+        if (m_scriptContext->GetConfig()->IsESBigIntEnabled())
+        {
+            likelyType = LikelyNumberType::BigInt;
+        }
+        *pdbl = Js::NumberUtilities::StrToDbl(p, &pchT, likelyType);
         Assert(pchT == p || !Js::NumberUtilities::IsNan(*pdbl));
```

Comment:
Can you add another Assert here that if the returned type is BigInt, then *pdbl == 0?

Reply:
will do